plugins/modules/site_workflow_manager.py

```diff
@@ -787,81 +1356,229 @@ def get_diff_merged(self, config):
+
+                else:
+                    site_exists, current_site = self.site_exists()
+                    if site_exists:
+                        self.created_site_list.append(site_name)
+                        self.log(
+                            "Site '{0}' created successfully".format(site_name), "INFO")
+
+            elif self.dnac_version >= self.version_2_3_7_6 and self.site_requires_update():
+                site_params = self.want.get("site_params")
+                site_params["site_id"] = self.have.get("site_id")
+                site_type = site_params.get("type")
+
+                response = (self.update_floor(site_params) if site_type == "floor"
+                            else self.update_area(site_params) if site_type == "area"
+                            else self.update_building(site_params) if site_type == "building"
+                            else self.log("Unknown site type: {0}".format(site_type), "ERROR"))
+
+                if response:
+                    self.log("Received API response from 'update_site': {0}".format(
+                        str(response)), "DEBUG")
+                    site_updated = True
+
+                if site_updated and response and isinstance(response, dict):
+                    taskid = response["response"]["taskId"]
+                    task_details = self.get_task_details(taskid)
+                    while True:
+                        if task_details.get("progress") == (
+                            "Group is updated successfully"
+                            if site_type != "floor"
+                            else "Service domain is updated successfully."
+                        ):
+                            self.result['changed'] = True
+                            self.result['response'] = task_details
+                            break
+                        elif task_details.get("bapiError"):
+                            self.module.fail_json(msg=task_details.get(
+                                "bapiError"), response=task_details)
+                            break
                 else:
-                    # Get the site id of the newly created site.
-                    (site_exists, current_site) = self.site_exists()
+                    self.update_not_neeeded_sites.append(site_name)
+                    self.log(
+                        "Site - {0} does not need any update".format(site_name), "INFO")
+                    return self
 
+        else:
+            self.log(site_updated)
+
+            if self.dnac_version <= self.version_2_3_5_3:
+                try:
+                    site_params = self.want.get("site_params")
+                    try:
+                        if site_params['site'].get('building'):
+                            building_details = {
+                                key: value for key, value in site_params['site']['building'].items() if value is not None}
```

Comment:
Avoid using multiple things in single line...code should be readable and easy.

Reply:
Modified